pkgs/tools/networking/wsl-vpnkit/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+
+# Runtime dependencies
+, gvproxy
+, dnsutils
+, gawk
+, iproute2
+, iptables
+, iputils
+, wget
+}:
+
+let
+  version = "0.4.1";
+  gvproxyCross = gvproxy.overrideAttrs (_: {
+    buildPhase = "make cross qemu-wrapper vm";
```

Comment:
Is this a common usecase? If so maybe an option or a top-level attr would be a good idea?

Reply:
I'm not sure, perhaps, I just checked what was publishing those binaries on their releases and it seems to be that command. Because `wsl-vpnkit` was using it from there.

Actually we only need the Windows binary, the vm can be used from the standard gvproxy.

For example this works as well (note this package is not the one from nixpkgs, but code is the same):
```patch
diff --git a/packages/wsl-vpnkit/default.nix b/packages/wsl-vpnkit/default.nix
index 807295c..ace87ad 100644
--- a/packages/wsl-vpnkit/default.nix
+++ b/packages/wsl-vpnkit/default.nix
@@ -12,8 +12,10 @@
   iputils,
   wget,
 }: let
-  gvproxyCross = gvproxy.overrideAttrs (_: {
-    buildPhase = "make cross qemu-wrapper vm";
+  gvproxyWin = gvproxy.overrideAttrs (_: {
+    buildPhase = ''
+      GOARCH=amd64 GOOS=windows go build -ldflags '-s -w' -o bin/gvproxy-windows.exe ./cmd/gvproxy
+    '';
   });
 in
   stdenv.mkDerivation rec {
@@ -31,8 +33,8 @@ in

     postPatch = ''
       substituteInPlace wsl-vpnkit \
-        --replace "/app/wsl-vm" "${gvproxyCross}/bin/vm" \
-        --replace "/app/wsl-gvproxy.exe" "${gvproxyCross}/bin/gvproxy-windows.exe"
+        --replace "/app/wsl-vm" "${gvproxy}/bin/vm" \
+        --replace "/app/wsl-gvproxy.exe" "${gvproxyWin}/bin/gvproxy-windows.exe"
     '';

     installPhase = ''
```

Reply:
I updated the PR to only build the gvproxy command for windows inside the `buildPhase`. And then I pick the gvforwarder (vm) from the gvproxy package directly.

Do you think it still makes sense to try to merge that into the main package?